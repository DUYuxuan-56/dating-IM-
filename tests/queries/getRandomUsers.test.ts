import { createTestClient } from 'apollo-server-testing';
import { gql } from 'apollo-server-express';
import { server } from '../../server';
import { resetDb } from '../../db';
import { mockAuth } from '../mocks/auth.provider';

describe('Query.getRandomUsers', () => {
  beforeEach(resetDb);

  it('should fetch random users', async () => {
    mockAuth(1);

    const { query } = createTestClient(server);

    let res = await query({
      query: gql`
        query GetRandomUsers {
          randomUsers {
            id
            name
            picture
          }
        }
      `,
    });

    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(res.data).toMatchSnapshot();
  });
});