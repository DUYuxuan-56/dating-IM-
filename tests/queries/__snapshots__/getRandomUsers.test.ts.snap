// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query.getRandomUsers should fetch random users 1`] = `
Object {
  "randomUsers": Array [],
}
`;
